vpr/src/place/net_cost_handler.cpp

```diff
@@ -1581,6 +1592,44 @@ double NetCostHandler::get_net_wirelength_from_layer_bb_(ClusterNetId net_id) {
     return ncost;
 }
 
+float NetCostHandler::get_chanz_cost_factor(const t_bb& bounding_box) {
+    float place_cost_exp = placer_opts_.place_cost_exp;
+    int x_high = bounding_box.xmax;
+    int x_low = bounding_box.xmin;
+    int y_high = bounding_box.ymax;
+    int y_low = bounding_box.ymin;
+
+    int num_inter_dir_conn;
+
+    if (x_low == 0 && y_low == 0) {
+        num_inter_dir_conn = acc_tile_num_inter_die_conn_[x_high][y_high];
+    } else if (x_low == 0) {
+        num_inter_dir_conn = acc_tile_num_inter_die_conn_[x_high][y_high] - \
+                             acc_tile_num_inter_die_conn_[x_high][y_low-1];
+    } else if (y_low == 0) {
+        num_inter_dir_conn = acc_tile_num_inter_die_conn_[x_high][y_high] - \
+                             acc_tile_num_inter_die_conn_[x_low-1][y_high];
+    } else {
+        num_inter_dir_conn = acc_tile_num_inter_die_conn_[x_high][y_high] - \
+                             acc_tile_num_inter_die_conn_[x_low-1][y_high] - \
+                             acc_tile_num_inter_die_conn_[x_high][y_low-1] + \
+                             acc_tile_num_inter_die_conn_[x_low-1][y_low-1];
+    }
```

Comment:
using `vtr::NdOffsetMatrix` will be consistent with thest of file and will get rid of this if statement.

Reply:
Yes, that would be cleaner.

Reply:
@soheilshahrouz, if you remember, my initial attempt was to use this data structure, but I encountered some memory issues with it. I’ll create an issue for this and, in another PR, attempt to replace this data structure with `vtr::NdOffsetMatrix`.

Reply:
I think I figured out what the problem was.
Check if 68ecb55 sovles the memory issue.

Reply:
Sounds good. I'll try it once the PR is merged.